### PR TITLE
Simplify/cleanup value factory manager in Ruby

### PR DESCRIPTION
**Pull request by bernardnormier**
also improves the Ice/exceptions test.

### CODE REVIEW

File: ruby/src/IceRuby/ValueFactoryManager.h [hunk 1]
```diff
@@ -69,8 +68,6 @@ namespace IceRuby
         VALUE _self;
         CustomFactoryMap _customFactories;
         const DefaultValueFactoryPtr _defaultFactory;
-
-        mutable std::mutex _mutex;
```
Comment (bernardnormier) [inline ruby/src/IceRuby/ValueFactoryManager.h:73]: All this code is single threaded, so don't need a mutex.

File: ruby/ruby/IceLocal/ValueFactory.rb [hunk 2]
```diff
@@ -1,28 +0,0 @@
...
-# Generated from file `ValueFactory.ice'
-#
-# Warning: do not edit this file.
-#
-# </auto-generated>
-#
-
-require 'Ice'
-
-module ::Ice
```
Comment (bernardnormier) [inline ruby/ruby/IceLocal/ValueFactory.rb:19]: This file didn't serve any purpose.

File: ruby/test/Ice/exceptions/AllTests.rb [hunk 3]
```diff
@@ -18,17 +18,25 @@ def allTests(helper, communicator)
     print "testing value factory registration exception... "
     STDOUT.flush
     vf = ValueFactoryI.new
```
Comment (bernardnormier) [inline ruby/test/Ice/exceptions/AllTests.rb:20]: See above: any class with a `create(id)` method is a ValueFactory.

File: ruby/src/IceRuby/ValueFactoryManager.cpp [hunk 4]
```diff
@@ -289,7 +272,7 @@ IceRuby_ValueFactoryManager_find(VALUE self, VALUE id)
 bool
 IceRuby::initValueFactoryManager(VALUE iceModule)
 {
-    _valueFactoryManagerClass = rb_define_class_under(iceModule, "ValueFactoryManagerI", rb_cObject);
+    _valueFactoryManagerClass = rb_define_class_under(iceModule, "ValueFactoryManager", rb_cObject);
```
Comment (bernardnormier) [inline ruby/src/IceRuby/ValueFactoryManager.cpp:275]: That's where we define the Ruby class Ice::ValueFactoryManager.

**Review by pepone — APPROVED** [review]

**Review by externl — APPROVED** [review]

**Pull request merged**